src/routes/Profile/WorkHistory/loader.ts

```diff
@@ -1,7 +1,15 @@
 import { client } from '@/api/client';
+import { authProvider } from '@/providers';
 import { LoaderFunction } from 'react-router-dom';
 
 export default (async ({ request }) => {
-  const { data } = await client.GET('/api/profiili/tyopaikat', { signal: request.signal });
-  return data;
+  try {
+    if (authProvider.loginState === 'loggedIn') {
+      const { data } = await client.GET('/api/profiili/tyopaikat', { signal: request.signal });
+      return data;
+    }
+  } catch (error) {
+    authProvider.expireSession();
+  }
+  return [];
```

Comment:
Added empty arrays to loaders as in some cases when going to these restricted pages the view's own loader still runs while not the view is not actually seen.